File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
    let cnt = 0;
    let limit = 0;
    for (let i = 1; i <= n; i++) {
        if (limit >= n) break;
        
        if (i % 2 === 1){
            let center = n / i;
            if(n % i === 0 && n === center * (2 * i - 1)) cnt++;
        } else {
            let center = Math.floor(n / i)
            if((i / 2) * (2 * center + 1) === n) cnt++;
        }
    }
    
    return cnt;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def recursive(k, _sum, n):
    if _sum > n:
        return False
    
    if _sum == n:
        return True
    
    return recursive(k + 1, _sum + k + 1, n)
        
    

def solution(n):
    cnt = 0
    li = []
    for i in range(1, n + 1):
        if recursive(i, i, n):
            li.append(i) 
            cnt += 1
            
    print(li)
    return cnt